File: android-sample/app/src/main/java/net/kibotu/bridgesample/bridge/commands/tracking/TrackScreenCommand.kt
package net.kibotu.bridgesample.bridge.commands.tracking

import net.kibotu.bridgesample.bridge.commands.BridgeCommand
import net.kibotu.bridgesample.bridge.commands.utils.BridgeParsingUtils
import net.kibotu.bridgesample.bridge.commands.utils.BridgeResponseUtils
import timber.log.Timber

/**
 * Tracks screen views (fire-and-forget).
 *
 * Separate from trackEvent because many analytics services treat page views
 * differently from events, and following Firebase Analytics conventions for
 * screen tracking with `screen_view` event name.
 *
 * Fire-and-forget: returns null so no response is sent back to web.
 */
class TrackScreenCommand : BridgeCommand {

    override val action = "trackScreen"

    override suspend fun handle(content: Any?): Any? {
        val screenName = BridgeParsingUtils.parseString(content, "screenName")
        if (screenName.isEmpty()) {
            return BridgeResponseUtils.createErrorResponse(
                "INVALID_PARAMETER",
                "Missing 'screenName' parameter"
            )
        }

        val screenClass = BridgeParsingUtils.parseString(content, "screenClass")

        Timber.i("[Bridge] Track screen: $screenName, class: $screenClass")

        // TODO: Forward to your analytics service (e.g. Firebase Analytics)

        return null
    }
}

File: ios/JSBridge/Sources/JSBridge/Commands/tracking/TrackScreenCommand.swift
import Foundation
import Orchard

/// Command for tracking screen views (fire-and-forget)
public final class TrackScreenCommand: BridgeCommand {
    public let action = "trackScreen"

    public init() {}

    @MainActor
    public func handle(content: [String: Any]?) async throws -> [String: Any]? {
        guard let screenName = content?["screenName"] as? String else {
            throw BridgeError.invalidParameter("screenName")
        }
        
        let screenClass = content?["screenClass"] as? String
        
        var parameters: [String: Any] = ["screen_name": screenName]
        if let screenClass = screenClass {
            parameters["screen_class"] = screenClass
        }
        
        let trackingEvent = BridgeScreenTrackingEvent(name: "screen_view", parameters: parameters)
        Orchard.v("\(trackingEvent)")
        Orchard.v("[Bridge] Track screen: \(screenName), class: \(String(describing: screenClass))")
        
        return nil
    }
}

private struct BridgeScreenTrackingEvent: TrackingEvent {
    let name: String
    let parameters: [String: Any]
}
